grpc/client.go

```diff
@@ -40,17 +43,28 @@ func ClientSetup(c *cmd.GRPCClientConfig, tlsConfig *tls.Config, metrics clientM
 		hnygrpc.UnaryClientInterceptor(),
 	}
 	allInterceptors = append(interceptors, allInterceptors...)
-	host, _, err := net.SplitHostPort(c.ServerAddress)
-	if err != nil {
-		return nil, err
+
+	if c.ServerAddress != "" {
+		host, _, err := net.SplitHostPort(c.ServerAddress)
+		if err != nil {
+			return nil, err
+		}
+		creds := bcreds.NewClientCredentials(tlsConfig.RootCAs, tlsConfig.Certificates, host)
+		return grpc.Dial(
+			"dns:///"+c.ServerAddress,
+			grpc.WithBalancerName("round_robin"),
+			grpc.WithTransportCredentials(creds),
+			grpc.WithChainUnaryInterceptor(allInterceptors...),
+		)
+	} else {
+		creds := bcreds.NewClientCredentials(tlsConfig.RootCAs, tlsConfig.Certificates, "")
+		return grpc.Dial(
+			"static:///"+strings.Join(c.ServerAddresses, ","),
+			grpc.WithBalancerName("round_robin"),
+			grpc.WithTransportCredentials(creds),
+			grpc.WithChainUnaryInterceptor(allInterceptors...),
+		)
```

Comment:
Because these two conditional clauses are so similar, can we instead have them just set a `hostOverride` and a `target`, and then have the `creds :=` and the `return grpc.Dial(target, grpc.With...)` be outside the conditional? Approximately like
```
var target string
var hostOverride string
if c.ServerAddress != "" {
  hostOverride, _, err := net.SplitHostPort(c.ServerAddress)
  target = "dns:///"+c.ServerAddress
} else {
  target = "static:///"+strings.Join(c.ServerAddresses, ",")
}
creds := bcreds.NewClientCredentials(..., hostOverride)
return grpc.Dial(target, ...)
```
This just makes it easier for a reader to quickly tell exactly what differs between the two branches, and ensures that things like the interceptors don't accidentally diverge in the future.